app/misc/repo.ts

```diff
@@ -258,11 +259,94 @@ function openRepository() {
     }
   }
 
+// works as a monitor for any change to the reference list
+function refreshList(verbose) {
+  Git.Repository.open(repoFullPath)
+    .then(function (repo) {
+      repo.getCurrentBranch()
+        .then(function () {
+          // grab the list
+          return repo.getReferences(Git.Reference.TYPE.LISTALL);
+        })
+        .then(function (refList) {
+          // sort refList alphabetically to get uniform order of the list
+          refList.sort();
```

Comment:
Since we're doing this every second, I would move the sorting until AFTER the comparison? Sorting is O(nlogn), the comparison is only O(n).

Reply:
@colinwerner sorting here is definitely an overhead. However for some reason, the refList here retrived were not always in consistent order, thus there needs a sorting before the comparison. otherwise there is likely a chance it refreshes the list when there is actually no change been made.

Reply:
@SiRumCz, thanks for the reply. I misread the code, you're right. The sorting is required.